server/master/src/main/java/org/apache/accumulo/master/tableOps/tableImport/MapImportFileNames.java

```diff
@@ -51,66 +51,69 @@
   @Override
   public Repo<Master> call(long tid, Master environment) throws Exception {
 
-    Path path = new Path(tableInfo.importDir, "mappings.txt");
-
-    BufferedWriter mappingsWriter = null;
-
-    try {
-      VolumeManager fs = environment.getFileSystem();
-
-      fs.mkdirs(new Path(tableInfo.importDir));
-
-      FileStatus[] files = fs.listStatus(new Path(tableInfo.exportDir));
-
-      UniqueNameAllocator namer = environment.getContext().getUniqueNameAllocator();
-
-      mappingsWriter = new BufferedWriter(new OutputStreamWriter(fs.create(path), UTF_8));
+    VolumeManager fs = environment.getFileSystem();
+    UniqueNameAllocator namer = environment.getContext().getUniqueNameAllocator();
+
+    for (ImportedTableInfo.DirectoryMapping dm : tableInfo.directories) {
```

Comment:
It would be good to verify that each export dir contains a disjoint set of file names.  This may be a good place to do this check.  If the check is not done and someone make mistake it seems like it could lead to multiple entries for same file, but I am not sure.

Reply:
Ok. I've been thinking about this too. Thanks for pointing out that this would be a good place to do this.